File: frontend/src/types/project.ts
/**
 * 一级项目相关类型定义
 */

import type { PaginationParams, PaginationResponse } from './api'

// 项目阶段
export type ProjectStage = 'sourcing' | 'screening' | 'due_diligence' | 'ic' | 'post_investment' | 'exit' | 'rejected'

// 项目行业
export type ProjectIndustry = 'tmt' | 'healthcare' | 'consumer' | 'manufacturing' | 'energy' | 'finance' | 'real_estate' | 'other'

// 跟进记录
export interface ProjectFollowUp {
  id: number
  project_id: number
  follow_date: string
  follow_type?: string
  content: string
  next_plan?: string
  follow_user_id?: number
  follow_user_name?: string
  created_at: string
}

// 阶段变更记录
export interface ProjectStageChange {
  id: number
  project_id: number
  from_stage?: ProjectStage
  to_stage: ProjectStage
  reason?: string
  operator_id?: number
  operator_name?: string
  created_at: string
}

// 项目
export interface Project {
  id: number
  project_code: string
  project_name: string
  short_name?: string
  industry?: ProjectIndustry
  sub_industry?: string
  source?: string
  source_channel?: string
  stage: ProjectStage
  assigned_user_id?: number
  assigned_user_name?: string
  // Sourcing
  initial_intro?: string
  contact_name?: string
  contact_phone?: string
  contact_email?: string
  // 初筛
  screening_date?: string
  screening_result?: string
  screening_notes?: string
  // 尽调
  dd_start_date?: string
  dd_end_date?: string
  dd_conclusion?: string
  // 投决
  ic_date?: string
  ic_result?: string
  investment_amount?: number
  valuation?: number
  shareholding_ratio?: number
  // 投后
  investment_date?: string
  board_seat?: boolean
  // 退出
  exit_method?: string
  exit_date?: string
  exit_amount?: number
  irr?: number
  moic?: number
  remark?: string
  created_at: string
  updated_at: string
  follow_ups?: ProjectFollowUp[]
  stage_changes?: ProjectStageChange[]
}


// 创建项目请求
export interface ProjectCreate {
  project_code: string
  project_name: string
  short_name?: string
  industry?: ProjectIndustry
  sub_industry?: string
  source?: string
  source_channel?: string
  stage?: ProjectStage
  assigned_user_id?: number
  initial_intro?: string
  contact_name?: string
  contact_phone?: string
  contact_email?: string
  remark?: string
}

// 更新项目请求
export interface ProjectUpdate extends Partial<Omit<ProjectCreate, 'project_code'>> {
  screening_date?: string
  screening_result?: string
  screening_notes?: string
  dd_start_date?: string
  dd_end_date?: string
  dd_conclusion?: string
  ic_date?: string
  ic_result?: string
  investment_amount?: number
  valuation?: number
  shareholding_ratio?: number
  investment_date?: string
  board_seat?: boolean
  exit_method?: string
  exit_date?: string
  exit_amount?: number
  irr?: number
  moic?: number
}

// 项目列表查询参数
export interface ProjectListParams extends PaginationParams {
  keyword?: string
  stages?: string
  industries?: string
  assigned_user_ids?: string
  sort_by?: string
  sort_order?: string
}

// 项目列表响应
export interface ProjectListResponse extends PaginationResponse<Project> {}

// 阶段流转请求
export interface StageTransfer {
  to_stage: ProjectStage
  reason?: string
}

// 跟进记录创建
export interface ProjectFollowUpCreate {
  follow_date: string
  follow_type?: string
  content: string
  next_plan?: string
}

// 项目统计
export interface ProjectStats {
  total: number
  by_stage: Record<string, number>
  by_industry: Record<string, number>
  total_investment: number
}

// 项目阶段选项
export const PROJECT_STAGE_OPTIONS = [
  { label: 'Sourcing', value: 'sourcing', color: '#909399' },
  { label: '初筛', value: 'screening', color: '#409EFF' },
  { label: '尽调', value: 'due_diligence', color: '#E6A23C' },
  { label: '投决', value: 'ic', color: '#F56C6C' },
  { label: '投后', value: 'post_investment', color: '#67C23A' },
  { label: '退出', value: 'exit', color: '#303133' },
  { label: '已否决', value: 'rejected', color: '#C0C4CC' }
] as const

// 项目行业选项
export const PROJECT_INDUSTRY_OPTIONS = [
  { label: 'TMT', value: 'tmt' },
  { label: '医疗健康', value: 'healthcare' },
  { label: '消费', value: 'consumer' },
  { label: '先进制造', value: 'manufacturing' },
  { label: '新能源', value: 'energy' },
  { label: '金融', value: 'finance' },
  { label: '房地产', value: 'real_estate' },
  { label: '其他', value: 'other' }
] as const

// 跟进方式选项
export const FOLLOW_TYPE_OPTIONS = [
  { label: '电话', value: '电话' },
  { label: '会议', value: '会议' },
  { label: '邮件', value: '邮件' },
  { label: '现场', value: '现场' },
  { label: '其他', value: '其他' }
] as const

// 退出方式选项
export const EXIT_METHOD_OPTIONS = [
  { label: 'IPO', value: 'IPO' },
  { label: '并购', value: '并购' },
  { label: '回购', value: '回购' },
  { label: '转让', value: '转让' },
  { label: '清算', value: '清算' },
  { label: '其他', value: '其他' }
] as const

File: backend/app/schemas/project.py
"""
一级项目Schema
"""
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import date, datetime
from enum import Enum


class ProjectStage(str, Enum):
    """项目阶段"""
    SOURCING = "sourcing"
    SCREENING = "screening"
    DUE_DILIGENCE = "due_diligence"
    IC = "ic"
    POST_INVESTMENT = "post_investment"
    EXIT = "exit"
    REJECTED = "rejected"


class ProjectIndustry(str, Enum):
    """项目行业"""
    TMT = "tmt"
    HEALTHCARE = "healthcare"
    CONSUMER = "consumer"
    MANUFACTURING = "manufacturing"
    ENERGY = "energy"
    FINANCE = "finance"
    REAL_ESTATE = "real_estate"
    OTHER = "other"


# ========== 跟进记录 ==========
class ProjectFollowUpBase(BaseModel):
    follow_date: date = Field(..., description="跟进日期")
    follow_type: Optional[str] = Field(None, description="跟进方式")
    content: str = Field(..., description="跟进内容")
    next_plan: Optional[str] = Field(None, description="下一步计划")


class ProjectFollowUpCreate(ProjectFollowUpBase):
    pass


class ProjectFollowUpResponse(ProjectFollowUpBase):
    id: int
    project_id: int
    follow_user_id: Optional[int]
    follow_user_name: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True


# ========== 阶段变更 ==========
class ProjectStageChangeResponse(BaseModel):
    id: int
    project_id: int
    from_stage: Optional[ProjectStage]
    to_stage: ProjectStage
    reason: Optional[str]
    operator_id: Optional[int]
    operator_name: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True


# ========== 项目 ==========
class ProjectBase(BaseModel):
    project_code: str = Field(..., description="项目编号")
    project_name: str = Field(..., description="项目名称")
    short_name: Optional[str] = Field(None, description="项目简称")
    industry: Optional[ProjectIndustry] = Field(None, description="行业")
    sub_industry: Optional[str] = Field(None, description="细分领域")
    source: Optional[str] = Field(None, description="项目来源")
    source_channel: Optional[str] = Field(None, description="来源渠道")
    stage: Optional[ProjectStage] = Field(ProjectStage.SOURCING, description="项目阶段")
    assigned_user_id: Optional[int] = Field(None, description="负责人ID")
    
    # Sourcing
    initial_intro: Optional[str] = Field(None, description="初步介绍")
    contact_name: Optional[str] = Field(None, description="联系人")
    contact_phone: Optional[str] = Field(None, description="联系电话")
    contact_email: Optional[str] = Field(None, description="联系邮箱")
    
    # 初筛
    screening_date: Optional[date] = None
    screening_result: Optional[str] = None
    screening_notes: Optional[str] = None
    
    # 尽调
    dd_start_date: Optional[date] = None
    dd_end_date: Optional[date] = None
    dd_conclusion: Optional[str] = None
    
    # 投决
    ic_date: Optional[date] = None
    ic_result: Optional[str] = None
    investment_amount: Optional[float] = None
    valuation: Optional[float] = None
    shareholding_ratio: Optional[float] = None
    
    # 投后
    investment_date: Optional[date] = None
    board_seat: Optional[bool] = False
    
    # 退出
    exit_method: Optional[str] = None
    exit_date: Optional[date] = None
    exit_amount: Optional[float] = None
    irr: Optional[float] = None
    moic: Optional[float] = None
    
    remark: Optional[str] = None


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    project_name: Optional[str] = None
    short_name: Optional[str] = None
    industry: Optional[ProjectIndustry] = None
    sub_industry: Optional[str] = None
    source: Optional[str] = None
    source_channel: Optional[str] = None
    assigned_user_id: Optional[int] = None
    initial_intro: Optional[str] = None
    contact_name: Optional[str] = None
    contact_phone: Optional[str] = None
    contact_email: Optional[str] = None
    screening_date: Optional[date] = None
    screening_result: Optional[str] = None
    screening_notes: Optional[str] = None
    dd_start_date: Optional[date] = None
    dd_end_date: Optional[date] = None
    dd_conclusion: Optional[str] = None
    ic_date: Optional[date] = None
    ic_result: Optional[str] = None
    investment_amount: Optional[float] = None
    valuation: Optional[float] = None
    shareholding_ratio: Optional[float] = None
    investment_date: Optional[date] = None
    board_seat: Optional[bool] = None
    exit_method: Optional[str] = None
    exit_date: Optional[date] = None
    exit_amount: Optional[float] = None
    irr: Optional[float] = None
    moic: Optional[float] = None
    remark: Optional[str] = None


class ProjectResponse(ProjectBase):
    id: int
    created_at: datetime
    updated_at: datetime
    assigned_user_name: Optional[str] = None
    follow_ups: List[ProjectFollowUpResponse] = []
    stage_changes: List[ProjectStageChangeResponse] = []
    
    class Config:
        from_attributes = True


class ProjectListResponse(BaseModel):
    id: int
    project_code: str
    project_name: str
    short_name: Optional[str]
    industry: Optional[ProjectIndustry]
    stage: Optional[ProjectStage]
    assigned_user_id: Optional[int]
    assigned_user_name: Optional[str] = None
    investment_amount: Optional[float]
    created_at: datetime
    
    class Config:
        from_attributes = True


class StageTransfer(BaseModel):
    """阶段流转"""
    to_stage: ProjectStage = Field(..., description="目标阶段")
    reason: Optional[str] = Field(None, description="流转原因")


class ProjectStats(BaseModel):
    """项目统计"""
    total: int
    by_stage: dict
    by_industry: dict
    total_investment: float
